build.sh

```diff
@@ -149,5 +150,40 @@ configure_user(){
     echo '%wheel ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers.d/wheel-nopasswd
 }
 
+write_repo() {
+    # Save in-container artifacts about the container for archiving in Koji.
+    mkdir -p /build
+    git_state="clean";
+    if [[ $(git diff --shortstat 2> /dev/null | tail -n1) != '' ]]; then
+        git_state="dirty"
+    fi
+
+    # We don't use git-archive here because submodules.
+    tarball="/build/src-archive.tar.gz"
+    tar -cvf ${tarball} --exclude-vcs /root/containerbuild/
+
+    # shellcheck disable=SC2046
+    cat > /build/meta.json <<EOC
+    cat <<EOC
+{
+    "arch": "$(uname -m)",
+    "build_date": "$(date -uR)",
+    "branch": "$(git symbolic-ref -q HEAD --short)",
+    "commit": "$(git rev-parse HEAD)",
+    "remote": "$(git remote get-url $(git for-each-ref --format='%(upstream:remotename)' $(git symbolic-ref -q HEAD)))",
```

Comment:
i'm trying to understand this line. You want to get the remote for each branch in the repo?

Reply:
Naw, that gives the upstream remote for the current branch. The output looks something like:
```
{
    "arch": "x86_64",
    "build_date": "Mon, 28 Jan 2019 17:14:19 +0000",
    "branch": "meta",
    "commit": "f07bc91b053f5a7aeaafc74847a92e4c02b94e8a",
    "remote": "https://github.com/darkmuggle/coreos-assembler",
    "state": "dirty",
    "src": {
        "file": "/build/src-archive.tar.gz",
        "format": "tar.gz",
        "size": "30433280",
        "checksum_type": "sha256",
        "checksum": "7723dd87f660126db5ca767a4178e50a987e5251015d8f15c4ff4f91da44935c",                                               
    }
}
```

Reply:
> The `git for-each-ref` gives the upstream remote for the current branch, by getting the ref for HEAD

👍